k8s-job/action.yml

```diff
@@ -33,7 +33,18 @@ runs:
           -n "$NAMESPACE" \
           --timeout="${JOB_TIMEOUT}" \
           "pods/${1}" && \
-        kubectl logs "pods/$1" -n "$NAMESPACE" --follow
+        # If the pod is placed on a node that just started (e.g. because the job triggered a scale up) and we try to
```

Comment:
Can you put this while loop into a function. 
Notice the command continuation here: `"pods/${1}" &&`
Ideally this is wrapped in a call so the continuation works correctly. 
Like so: `"pods/${1}" && safe_logs $1`
This would help with readability too.

Reply:
For sure, updated and retested. Same results as in the descriptions (both with and without the race condition).